packages/block-library/src/list/v2/edit.js

```diff
@@ -14,20 +15,53 @@ import {
 	formatListNumbered,
 	formatListNumberedRTL,
 } from '@wordpress/icons';
+import { useEffect } from '@wordpress/element';
+import { useRegistry, useDispatch } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import OrderedListSettings from '../ordered-list-settings';
+import { migrateToListV2 } from './migrate';
 
 const TEMPLATE = [ [ 'core/list-item' ] ];
 
-function Edit( { attributes, setAttributes } ) {
+/**
+ * At the moment, deprecations don't handle create blocks from attributes
+ * (like when using CPT templates). For this reason, this hook is necessary
+ * to avoid breaking templates using the old list block format.
+ *
+ * @param {Object} attributes Block attributes.
+ * @param {string} clientId   Block client ID.
+ */
+function useMigrateOnLoad( attributes, clientId ) {
+	const registry = useRegistry();
+	const { updateBlockAttributes, replaceInnerBlocks } = useDispatch(
+		blockEditorStore
+	);
+
+	useEffect( () => {
+		// As soon as the block is loaded, migrate it to the new version.
+
+		if ( ! attributes.values ) {
+			return;
+		}
+		const [ newAttributes, newInnerBlocks ] = migrateToListV2( attributes );
+
+		registry.batch( () => {
```

Comment:
Is there a solution to avoid creating an undo level?

Reply:
Not that I can think of, but I think it's fine here, I also considered adding a "deprecated" call.